### PR TITLE
improve additional filters support

### PR DESCRIPTION
**Pull request by sensorario**
| Q             | A
| ------------- | ---
| Branch?       | 2.2
| Bug/Hotfix?   |  no
| Refactoring?  | no
| New feature?  | yes
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes

This PR aims to move inside QueryBundle all components necessary to apply additional filters.

### CODE REVIEW

File: src/Mado/QueryBundle/Interfaces/CustomFilter.php [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+        RequestStack $requestStack,
+        IdsChecker $idsChecker,
+        LoggerInterface $logger
+    );
+
+    public function setUser(AdditionalFilterable $user);
+
+    public function allItemsTo(string $entity);
+
+    public function getPathTo(string $fullyqualifiedClassName);
```
Comment (AlessandroMinoccheri) [inline src/Mado/QueryBundle/Interfaces/CustomFilter.php]: change this:
fullyqualifiedClassName

to this:
fullyQualifiedClassName

File: src/Mado/QueryBundle/Services/IdsChecker.php [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+    public function validateIds()
+    {
+        $rawFilteredIds = $this->additionalFilter->getIds();
+
+        foreach ($this->filtering as $key => $queryStringIds) {
+            $querystringIds = explode(',', $queryStringIds);
+            $additionalFiltersIds = explode(',', $this->additionalFilter->getIds());
+            foreach ($querystringIds as $requestedId) {
+                if ($this->additionalFilter->getOperator() == 'list') {
+                    if (!in_array($requestedId, $additionalFiltersIds)) {
```
Comment (AlessandroMinoccheri) [inline src/Mado/QueryBundle/Services/IdsChecker.php]: is possible to move some piece of code into other functions or objects?

File: src/Mado/QueryBundle/Interfaces/CustomFilter.php (hunk 1 above)
Comment (sensorario) [reply src/Mado/QueryBundle/Interfaces/CustomFilter.php]: 👍

File: src/Mado/QueryBundle/Services/IdsChecker.php (hunk 2 above)
Comment (sensorario) [reply src/Mado/QueryBundle/Services/IdsChecker.php]: This class is already the result of a refactoring. I am afraid that another decoupling, now, could be an over engineered solution. Yes, it is possibile. Not so useful now.

**Pull request merged**